lib/java/com/google/android/material/bottomsheet/BottomSheetDialog.java

```diff
@@ -112,6 +115,54 @@ protected void onStart() {
     }
   }
 
+  /**
+   * This function can be called from a few different use cases, including Swiping the dialog down
+   * or calling `dismiss()` from a `BottomSheetDialogFragment`, tapping outside a dialog, etc...
+   *
+   * The default animation to dismiss this dialog is a fade-out transition.
+   * We want to utilize the swipe down dismiss animation for all cases.
+   *
+   * If this function is called from a swipe down interaction, then keep the default behavior.
+   *
+   * Else, since this is a terminal event which will finish this dialog, we override the attached
+   * {@link com.google.android.material.bottomsheet.BottomSheetBehavior.BottomSheetCallback}
+   * to call this function, after {@link BottomSheetBehavior#STATE_HIDDEN} is set.
+   * This will enforce the swipe down animation before canceling this dialog.
+   */
+  @Override
+  public void cancel() {
+    if (getBehavior().getState() == BottomSheetBehavior.STATE_HIDDEN) {
+      super.cancel();
+    } else {
+      BottomSheetBehavior behavior = getBehavior();
+
+      // If the default callback was not overridden, then set the state to STATE_HIDDEN and defer
+      // to the default callback behavior.
+      if (behavior.getBottomSheetCallback() == bottomSheetCallback) {
+        behavior.setState(BottomSheetBehavior.STATE_HIDDEN);
+      } else {
+        WeakReference<BottomSheetDialog> dialogWeakReference = new WeakReference<>(this);
+
+        behavior.setBottomSheetCallback(new BottomSheetBehavior.BottomSheetCallback() {
```

Comment:
Would it work to just set the callback back to `bottomSheetCallback`? It does the same thing right?

Reply:
Yep that work.
Just pushed a commit to address that 👍